app/packages/core/src/components/Filters/categoricalFilter/filterOption/FilterOption.tsx

```diff
@@ -0,0 +1,301 @@
+  const options: Option[] = [];
+  if (Boolean(nestedField)) {
+    options.push({
+      icon: "FilterAltIcon",
+      key: "filter",
+      value: `Select ${nestedField} with ${valueName}`,
+      tooltip: "dataset.filter_labels(field, expr, only_matches=True)",
+    });
+  }
+  if (Boolean(nestedField) && !shouldNotShowExclude) {
+    options.push({
+      icon: "FilterAltOffIcon",
+      key: "negativeFilter",
+      value: `Exclude ${nestedField} with ${valueName}`,
+      tooltip: "dataset.filter_labels(field, ~expr, only_matches=False)",
+    });
+  }
+  if (!modal && !isKeyPointLabel) {
+    options.push({
+      icon: "ImageIcon",
+      key: "match",
+      value: `Show samples with ${valueName}`,
+      tooltip: Boolean(nestedField)
+        ? "dataset.match_labels(fields=field, filter=expr)"
+        : "dataset.match(expr)",
+    });
+  }
+  if (!modal && !shouldNotShowExclude && !isKeyPointLabel) {
+    options.push({
+      icon: "HideImageIcon",
+      key: "negativeMatch",
+      value: `Omit samples with ${valueName}`,
+      tooltip: Boolean(nestedField)
+        ? "dataset.match_labels(fields=field, filter=expr, bool=False)"
+        : "dataset.match(~expr)",
+    });
+  }
+  return options;
+};
+
+const Text = styled.div`
+  font-size: 1rem;
+  margin: auto auto auto 5px;
+  ${({ theme }) => theme.text.secondary};
+`;
+
+const FilterOptionContainer = styled.div`
+  position: relative;
+`;
+
+const FilterOption: React.FC<Props> = ({
+  color,
+  modal,
+  isKeyPointLabel,
+  valueName,
+  nestedField,
+  shouldNotShowExclude,
+  excludeAtom,
+  onlyMatchAtom,
+  isMatchingAtom,
+}) => {
+  const [key, setKey] = React.useState<Key | null>(null);
+
+  const [open, setOpen] = React.useState(false);
+  const [excluded, setExcluded] = useRecoilState(excludeAtom);
+  const [onlyMatch, setOnlyMatch] = useRecoilState(onlyMatchAtom);
+  const [isMatching, setIsMatching] = useRecoilState(isMatchingAtom);
+
+  const theme = useTheme();
+  const highlightedBGColor = Color(color).alpha(0.25).string();
+
+  const popoutRef = React.useRef();
+  const ref = React.useRef();
+
+  useOutsideClick(popoutRef, () => {
+    setOpen(false);
+  });
+
+  const options = generateOptions(
+    nestedField,
+    shouldNotShowExclude,
+    modal,
+    isKeyPointLabel,
+    valueName
+  );
+
+  useEffect(() => {
+    // on initial load, if filter already exists, load exisiting filter for modal filters
+    // otherwise, show defaults: filter for nested listfield, match for other fields
+    if (key === null) {
+      // when filter initializes, isMatching is defaulted to false, but we need to use Boolean(nestedField) to set normal defaults to match
+      if (!excluded) {
+        Boolean(nestedField) ? setKey("filter") : setKey("match");
+      } else {
+        Boolean(nestedField)
+          ? setKey("negativeFilter")
+          : setKey("negativeMatch");
+      }
+    }
+  }, []);
```

Comment:
nestedField and excluded are dependencies here. They probably won't change, but unforeseen changes my change that and cause issues
```suggestion
  }, [nestedField, excluded]);
```